commands.go

```diff
@@ -110,7 +110,16 @@ var Commands = []cli.Command{
 
 			host, err := store.Create(name, driver, c)
 			if err != nil {
-				log.Fatal(err)
+				log.Errorf("Error creating host: %s", err)
+				if c.GlobalBool("debug") {
+					log.Fatal(err)
+				} else {
+					log.Warnf("Removing provisioned machine.  Use --debug to prevent removal.")
```

Comment:
IMO: "Removing created machine.  You can run machine with the --debug flag to avoid this."